include/zephyr/sys/util.h

```diff
@@ -202,6 +202,23 @@ extern "C" {
 		(POINTER_TO_UINT(ptr) - POINTER_TO_UINT(array)) / sizeof((array)[0]);              \
 	})
 
+/**
+ * @brief Iterate over members of an array using an index variable
+ *
+ * @param array the array in question
+ * @param idx name of array index variable
+ */
+#define ARRAY_FOR_EACH(array, idx) for (size_t idx = 0; (idx) < ARRAY_SIZE(array); ++(idx))
+
+/**
+ * @brief Iterate over members of an array using a pointer
+ *
+ * @param array the array in question
+ * @param ptr pointer to an element of @p array
+ */
+#define ARRAY_FOR_EACH_PTR(array, ptr)                                                             \
+	for (__typeof__(*(array)) *ptr = (array); ((ptr) - (array)) < ARRAY_SIZE(array); ++(ptr))
```

Comment:
Should add tests for these under `tests/unit/util`.

Reply:
Done!

Reply:
Thanks.

By the way, the first two commits in this PR say "(cherry picked from commit ...)," referencing commits in a fork.

Reply:
Fixed